core/src/Revolution/modElement.php

```diff
@@ -726,19 +726,14 @@ public function getPropertySet($setName = null)
     {
         $propertySet = null;
         $name = $this->get('name');
-        if (strpos($name, '@') !== false) {
-            $psName = '';
-            $split = xPDO:: escSplit('@', $name);
-            if ($split && isset($split[1])) {
-                $name = $split[0];
-                $psName = $split[1];
-                $filters = xPDO:: escSplit(':', $setName);
-                if ($filters && isset($filters[1]) && !empty($filters[1])) {
-                    $psName = $filters[0];
-                    $name .= ':' . $filters[1];
-                }
-                $this->set('name', $name);
-            }
+
+        $nameSplit = explode(':', $name);
+        $tagName = array_shift($nameSplit);
+        $remainingTag = implode(':', $nameSplit);
+        if (strpos($tagName, '@') !== false) {
+            $split = xPDO:: escSplit('@', $tagName);
+            $psName = $split[1];
+            $this->set('name', $split[0] . $remainingTag);
```

Comment:
Note that this is where the issue lies (same goes in the modTag file L520). Here, when modifiers are present, you end up with ``snippetNamedefault=`default val` `` instead of  ``snippetName:default=`default val` `` (note the colon missing in the former) when setting the name on the Element.

Reply:
A couple other things:
- I made my own submission for this (#16682) only because it was easier to do, particularly because I wanted to make sure whatever I came up with passed on the latest 3.1 dev branch _and_ I've been having issues getting the unit tests to run on my local dev install.
- I refactored how the two parts of the tag are broken up (using substr approach instead of explode/implode), as it seemed easier to see what was happening